### PR TITLE
[WIP] Debug Symbol

### PR DESCRIPTION
**Pull request by sklam**
* Add minimal support for generating debug info for functions
* Support GDB version >7
* Require LLVM build with debug symbol
     * This is due to how GDB find out about JIT-ed function.  It needs to recognize a function call by looking at debug info.
* Require MCJIT with jit_debug=True

### CODE REVIEW

**Comment by sklam** [comment]
**Note**
Need to build LLVM with the flags in http://llvm.org/docs/Packaging.html#compile-flags

**Comment by pitrou** [comment]
Can you give pointers to the docs you used? That would be useful :)

File: llvmlite/ir/debuginfo.py [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+    DW_TAG_base_type = 36
+    DW_TAG_array_type = 1
+    DW_TAG_enumeration_type = 4
+    DW_TAG_structure_type = 19
+    DW_TAG_union_type = 23
+    DW_TAG_inheritance = 28
+
+
+class DI_Attribute(IntEnum):
+    DW_AT_sibling = 0x01,
```
Comment (pitrou) [inline llvmlite/ir/debuginfo.py]: Are those deliberately tuples? Just asking.

Comment (sklam) [inline llvmlite/ir/debuginfo.py]: oops, no. good catch

**Comment by sklam** [comment]
I am using documentation from: http://llvm.org/docs/SourceLevelDebugging.html
But that is not enough information.  A lot is from trial and error and comparing against what clang is doing.

**Comment by sklam** [comment]
I think this PR is mostly a demonstration that debug info does work on some platform.

Let's move to 3.6 without merging this and I will update the PR afterwards.  Does 3.6 has the new metadata where they look like namedtuples?  That makes debugging the debuginfo a lot easier.

**Comment by pitrou** [comment]
Hmm, actually it seems the new metadata is only available on 3.7: http://llvm.org/docs/LangRef.html#specialized-metadata-nodes

**Comment by whitequark** [comment]
This can be closed, now that #162 is fixed.

**Comment by pitrou** [comment]
Right, closing.

**Pull request closed without merging**